packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx

```diff
@@ -407,6 +407,18 @@ describe('<DateField /> - Editing', () => {
       });
     });
 
+    it('should support 2-digits year format when a value is provided', () => {
+      testChange({
+        format: 'yy', // This format is not present in any of the adapter formats
+        defaultValue: adapterToUse.date(new Date(2022, 5, 4)),
+        keyStrokes: [
+          { value: '2', expected: '02' },
```

Comment:
Shouldn't it also support `0` as the first keystroke?
I've tried entering `02`, but end up with only `2` in this case 😞

Reply:
Did you test in the doc or in a test env ?
And in both cases, with which adapter ?
Because maybe the format just does not have trailing zeroes.

Reply:
> Did you test in the doc or in a test env ?

Doc

> And in both cases, with which adapter ?

Just the BasicFieldExample with `DD/MM/YY` format

Reply:
It is probably a different topic, but wouldn't you say that it ties together with 2-digit editing? 🤔

Reply:
I think I mis-understood your last message.

The feature I was referring to is the ability to type `02` on a 4 digit year and have it target `2002` instead of `0002` (like it does on the native date object).

But I think I now understand what you mean.
On a 2 digit year, pressing `02` marks `2` and sets the date to `0002` when it should set it to `2002`.
I'll have a look on how to fix that one.

Reply:
Looks like I misinterpreted the bug in the 1st place
The issue was that with a 2-digit year, my detection code to know if there was a trailing zero was broken.
So the section value generated was wrong (`2` instead of `02`) and `dayjs` is super strict on that aspect.

Looks like just fixing this logic works.

Reply:
We should probably add a case asserting the newly added behaviour of supporting 2 digit year entry with preceding zero. 🤔

Reply:
The format used in the 2 digit test has a trailing zero.
And it's the only 2 digit year format on `date-fns` :/

We could add an example with `dayjs`

I think we also need to improve our testing setup to easily choose on which adapter we are running the tests.
To avoid running everything with each adapter.
But to cover some edge-case and run all the heavily adapter-dependent tests on the 4 main adapters.